controller/src/beerocks/bml/bml.cpp

```diff
@@ -709,3 +709,44 @@ int bml_start_dcs_single_scan(BML_CTX ctx, const char *radio_mac, int dwell_time
     return pBML->start_dcs_single_scan(tlvf::mac_from_string(std::string(radio_mac)), dwell_time,
                                        channel_pool, channel_pool_size);
 }
+
+int bml_client_get_client_list(BML_CTX ctx, char *client_list, unsigned int *client_list_size)
+{
+    // Validate input parameters
+    if (!ctx && !client_list && !client_list_size) {
+        return (-BML_RET_INVALID_ARGS);
+    }
+
+    auto pBML = static_cast<bml_internal *>(ctx);
+    std::string temp_client_list;
+    int ret = pBML->client_get_client_list(temp_client_list, client_list_size);
+    if (!ret) {
+        beerocks::string_utils::copy_string(client_list, temp_client_list.c_str(),
+                                            *client_list_size);
+    }
```

Comment:
`client_list_size` is an IN/OUT param. As an input, it supposed to tell the size of the `client_list`.    
Following the code, the value of `client_list_size` is changed in the `pBML->client_get_client_list()` function. so you don't protect against  overflow of `client_list`   
```suggestion
    int ret = pBML->client_get_client_list(temp_client_list);
    if (!ret &&  *client_list_size <= temp_client_list.size()) {
        beerocks::string_utils::copy_string(client_list, temp_client_list.c_str(),
                                            *client_list_size);
        *client_list_size = temp_client_list.size();
    } else ret = ERROR;
```

Reply:
Added the protection in bml_internal, since the value of `client_list_size` is changed there